MapboxCoreNavigation/RouteController.swift

```diff
@@ -719,23 +721,31 @@ extension RouteController: CLLocationManagerDelegate {
                 
                 // Look at the following step to determine what the new alert level should be
                 if let upComingStep = routeProgress.currentLegProgress.upComingStep {
-                    alertLevel = upComingStep.expectedTravelTime <= RouteControllerMediumAlertInterval ? .medium : .low
+                    alertLevel = calculateNewAlert(from: upComingStep)
                 } else {
                     assert(false, "In this case, there should always be an upcoming step")
                 }
             }
-        } else if secondsToEndOfStep <= RouteControllerHighAlertInterval && routeProgress.currentLegProgress.currentStep.distance > minimumDistanceForHighAlert {
```

Comment:
@1ec5 @frederoni @ericrwolfe I'm removing the `minimumDistanceForMediumAlert` heuristics to keep this logic cleaner and focused only on seconds remaining. Also see [constants diff](https://github.com/mapbox/mapbox-navigation-ios/pull/448/files#diff-1e0aeb18d31c8c1abfce278ab3489151L121).